lib/src/database/sqflite_box.dart

```diff
@@ -0,0 +1,274 @@
+import 'dart:convert';
+
+import 'package:sqflite_common/sqlite_api.dart';
+
+/// Key-Value store abstraction over Sqflite so that the sdk database can use
+/// a single interface for all platforms. API is inspired by Hive.
+class BoxCollection {
+  final Database _db;
+  final Set<String> boxNames;
+
+  BoxCollection(this._db, this.boxNames);
+
+  static Future<BoxCollection> open(
+    String name,
+    Set<String> boxNames, {
+    int version = 1,
+    Object? sqfliteDatabase,
+    dynamic idbFactory,
+  }) async {
+    if (sqfliteDatabase is! Database) {
+      throw ('You must provide a Database `sqfliteDatabase` for FluffyBox on native.');
+    }
+    final batch = sqfliteDatabase.batch();
+    for (final name in boxNames) {
+      batch.execute(
+        'CREATE TABLE IF NOT EXISTS $name (k TEXT PRIMARY KEY NOT NULL, v TEXT)',
```

Comment:
The STRICT mode is only available since sqlite 3.37.0 which has been released at 2021-11-27. Not sure if we can garantuee that all devices support it. If a device does not support it, the create table action would fail

Reply:
Hm, I see. I guess it doesn't give us much of a benefit here anyway.